neo/tools/debugger/DebuggerServer.cpp

```diff
@@ -241,43 +272,59 @@ Send a message with no data to the debugger server.
 */
 void rvDebuggerServer::SendMessage ( EDebuggerMessage dbmsg )
 {
-	msg_t	 msg;
+	idBitMsg	 msg;
 	byte	 buffer[MAX_MSGLEN];
 
-	MSG_Init( &msg, buffer, sizeof( buffer ) );
-	MSG_WriteShort ( &msg, (int)dbmsg );
+	msg.Init( buffer, sizeof( buffer ) );
+	msg.BeginWriting();
+	msg.WriteShort ( (short)dbmsg );
 
-	SendPacket ( msg.data, msg.cursize );
+	SendPacket ( msg.GetData(), msg.GetSize() );
 }
 
 /*
 ================
 rvDebuggerServer::HandleAddBreakpoint
 
 Handle the DBMSG_ADDBREAKPOINT message being sent by the debugger client.  This
-message is handled by adding a new breakpoint to the breakpoint list with the
+message is handled by first checking if it is valid
+and is added as a new breakpoint to the breakpoint list with the
 data supplied in the message.
 ================
 */
-void rvDebuggerServer::HandleAddBreakpoint ( msg_t* msg )
+void rvDebuggerServer::HandleAddBreakpoint ( idBitMsg* msg )
 {
 	bool onceOnly = false;
 	long lineNumber;
 	long id;
-	char filename[MAX_PATH];
+	char filename[2048]; // DG: randomly chose this size
 
 	// Read the breakpoint info
-	onceOnly   = MSG_ReadBits ( msg, 1 ) ? true : false;
-	lineNumber = MSG_ReadInt ( msg );
-	id		   = MSG_ReadInt ( msg );
+	onceOnly   = msg->ReadBits ( 1 ) ? true : false;
```

Comment:
(damn, I wrote this last night but forgot to submit the review - maybe I should stick to normal comments)

Reply:
will remove it

Reply:
does the debugger client not offer a "break only once" functionality that is supposed to be communicated with this bit?
in that case it should probably be implemented instead of removed? or is it implemented already just differently?

Reply:
I am not sure why this is here. it is not used, and on GLP release it has never been functional. but that last bit doesnt say much since it wasnt functional all togheter. 
Anyway, maybe it was used originally to determine the next breapoint that should hit when you pressed pause? @motorsep do you have an idea?

Reply:
Maybe it was meant to be used with the "Run To Cursor" feature which doesn't seem to work?
i.e. create an only-once breakpoint at the cursor position and continue - it would break there and automatically remove the breakpoint

Reply:
Don't know , also doesn't seem important enough to do something with it.